src/java/picard/vcf/LiftoverVcf.java

```diff
@@ -0,0 +1,232 @@
+package picard.vcf;
+
+import htsjdk.samtools.Defaults;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.liftover.LiftOver;
+import htsjdk.samtools.reference.ReferenceSequenceFileWalker;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.CollectionUtil;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+import htsjdk.samtools.util.SequenceUtil;
+import htsjdk.samtools.util.SortingCollection;
+import htsjdk.samtools.util.StringUtil;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import htsjdk.variant.variantcontext.writer.Options;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder;
+import htsjdk.variant.vcf.VCFFileReader;
+import htsjdk.variant.vcf.VCFFilterHeaderLine;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFRecordCodec;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.CommandLineProgramProperties;
+import picard.cmdline.Option;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.VcfOrBcf;
+
+import java.io.File;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tool for lifting over a VCF to another genome build and producing a properly header'd,
+ * sorted and indexed VCF in one go.
+ *
+ * @author Tim Fennell
+ */
+@CommandLineProgramProperties(
+        usage = "Lifts a VCF over from one genome build to another using UCSC liftover. The output file will be sorted " +
+                "and indexed. Records may be rejected because they cannot be lifted over or because post-liftover the " +
+                "reference allele mismatches the target genome build.  Rejected records will be emitted with filters " +
+                "to the REJECT file, on the source genome.",
+        usageShort = "Lifts a VCF between genome builds.",
+        programGroup = VcfOrBcf.class
+)
+public class LiftoverVcf extends CommandLineProgram {
+    @Option(shortName=StandardOptionDefinitions.INPUT_SHORT_NAME, doc="The input VCF/BCF file to be lifted over.")
+    public File INPUT;
+
+    @Option(shortName=StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc="The output location to write the lifted over VCF/BCF to.")
+    public File OUTPUT;
+
+    @Option(shortName="C", doc="The liftover chain file.")
```

Comment:
perhaps link to a format spec for this file? https://genome.ucsc.edu/goldenPath/help/chain.html